src/common/server.ts

```diff
@@ -48,6 +49,54 @@ function executeCommand(command: string): Promise<string> {
   });
 }
 
+type VersionInfo = {
+  major: number;
+  minor: number;
+  patch: number;
+};
+
+/**
+ * Convert a version object to a string.
+ */
+function versionToString(version: VersionInfo): string {
+  return `${version.major}.${version.minor}.${version.patch}`;
+}
+
+/**
+ * The minimum version of the Ruff executable that supports the native server.
+ */
+const MINIMUM_RUFF_SERVER_VERSION: VersionInfo = { major: 0, minor: 3, patch: 5 };
```

Comment:
This comes from https://github.com/astral-sh/ruff-vscode/pull/443 although the minimum supported version if actually `0.3.3` but I guess this is the version we moved away from using the bundled version.

Reply:
As explained in https://github.com/astral-sh/ruff-vscode/issues/508 there are also commands/features that unsupported by old version of nativeServer and/or ruff-lsp. should we declare version constant for them? e.g. `ruff.printDebugInformation` which was added in https://github.com/astral-sh/ruff/pull/11831 for nativeServer

Reply:
Good point. I intend to implement the warning part for the settings in https://github.com/astral-sh/ruff-vscode/pull/505.

> `ruff.printDebugInformation` which was added in [astral-sh/ruff#11831](https://github.com/astral-sh/ruff/pull/11831) for nativeServer

Regarding this command, I think we should only register a VS Code command only if it's available in the server capabilities (https://github.com/astral-sh/ruff/pull/11850). This way we don't need to check the Ruff version for at least the server capabilities.

Reply:
I've opened #513 to keep track of it.